### PR TITLE
PLT-567 Replaced Dubious BigDecimal Scaling and Formatting

### PR DESCRIPTION
**Pull request by oliver-richers**
- automated testing: extended `DemoInvoicesTransientFieldFilteringTest`
- manual testing: open the DemoInvoicePage and test filtering

### CODE REVIEW

File: platform-emf/src/main/java/com/softicar/platform/emf/data/table/filter/value/EmfDataTableValueFilter.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ private IDisplayString getValueDisplayString() {
 		if (value instanceof IDisplayable) {
 			return ((IDisplayable) value).toDisplay();
 		} else {
-			return IDisplayString.create(" " + value);
+			return IDisplayString.create("" + value);
```
Comment (oliver-richers) [inline platform-emf/src/main/java/com/softicar/platform/emf/data/table/filter/value/EmfDataTableValueFilter.java:59]: unrelated change: this space was redundant

File: platform-dom/src/main/java/com/softicar/platform/dom/elements/number/decimal/DomBigDecimalInput.java [hunk 2]
```diff
@@ -4,20 +4,17 @@
...
+import java.util.Optional;
 
 /**
  * A {@link DomTextInput} that verifies that the user entered a valid
  * {@link BigDecimal} value.
  *
  * @author Semsudin Mekanic
+ * @author Oliver Richers
  */
 public class DomBigDecimalInput extends AbstractDomNumberInput {
```
Comment (oliver-richers) [inline platform-dom/src/main/java/com/softicar/platform/dom/elements/number/decimal/DomBigDecimalInput.java:16]: this input now does not force any scaling on the user

Comment (oliver-richers) [reply platform-dom/src/main/java/com/softicar/platform/dom/elements/number/decimal/DomBigDecimalInput.java:16]: image

**Review by alexander-schmidt — APPROVED** [review]
